packages/upswyng-server/EVENTLOG.md

```diff
@@ -0,0 +1,154 @@
+      alertTitle: alert.title,
+    },
+    kind: "alert_live",
+  }).save();
+} catch (e) {
+  console.error(
+    `Error creating Event Log \`alert_live\` for Alert ${_id}: ${e}`
+  );
+}
+```
+
+### Modify UI to display the Event Log
+
+The code behind the Event Logs page fetches recent Event Logs and displays a timeline
+where each Event Log is rendered by `EventLogItem.svelte`. The component contains a wrapper
+`div` around `if`-`else` statements which check the `kind`. We'll add a branch to the `if`-`else`s
+for our `kind`. We can copy-paste the `.timeline-marker` `div` and the `.timeline-content.heading`
+from one of the other `kind`s and pick an icon from [FontAwesome](https://www.fontawesome.com)
+to represent our Event Log `kind`. The second `p` of `div.timeline-content` will be specific to our
+Event Log. We'll just show the title and a link to the Alert, but some Event Logs will require
+more. Check out the "Event Log" page and the code to render some of the other `kind`s for ideas.
+
+Once done, the `EventLogItem` component body will look like:
+
+```html
+<div class="timeline-item">
+  {#if eventLog.detail.kind === 'some_event_log_kind'}
+    <!-- ... -->
+  {:else if eventLog.detail.kind === 'alert_live'}
+    <div class="timeline-marker is-icon">
+      <i class="fas fa-exclamation" />
+    </div>
+    <div class="timeline-content">
+      <p class="heading">
+        {#if timeAgo}{timeAgo.format(new Date(eventLog.createdAt))}{/if}
+      </p>
+      <p>
+        Alert
+        <a href={`/alert?id=${eventLog.detail.alertId}`} rel="prefetch">
+          <span class="has-text-weight-semibold">
+            {eventLog.detail.alertTitle}
+          </span>
+        </a>
+        is now live
+      </p>
+    </div>
+  {:else if eventLog.detail.kind === 'another_kind'}
+    <!-- ... -->
+  {/if}
+</div>
+```
+
+### Make `upswyngbot` message the Slack group
+
+There are just two simple steps to do to enable messaging to Slack.
+
+First we'll add a `case` to the `switch` in the `createTextForEventLog`
+function in `utility/slackbot.ts` to translate our Event Log to the
+message that will be sent to Slack. Check out some of the other `case`s to get an idea of what the logic should look like. Note that if you want to add an emoji you use the Slack format `:<emoji-name>:`. Check out the Slack docs [here](https://www.livexlive.com/live-events/festival/music-lives?utm_source=twitter&utm_medium=social&utm_campaign=event-musiclives2020). One other note: Slack uses [custom markup](https://api.slack.com/reference/surfaces/formatting) to add style and links to text.
+
+Next, add a call to `postEventLogMessage` from `utility/slackbot.ts` where appropriate, most commonly just after where you created and saved the Event Log.
+
+> If you don't want `upswyngbot` to send a message to Slack for your event you still need to add a `case`\
+> to the `switch` statement from above. It should throw an error because it shouldn't ever get called.
```

Comment:
So are you saying that if I create an event that I don't care to send to Slack, you still need to add a case that prevents the default throwing an error? If so, what should the do-nothing case look like?

Reply:
throw an error